binding.cc

```diff
@@ -1015,6 +1025,86 @@ namespace zmq {
 
 #endif
 
+  NAN_METHOD(Socket::Readv) {
+    Socket* socket = GetSocket(info);
+    if (socket->state_ != STATE_READY)
+      return info.GetReturnValue().Set(Nan::Undefined());
+
+    int events;
+    size_t events_size = sizeof(events);
+
+    int rc = 0;
+    int flags = 0;
+    int64_t more = 1;
+    size_t more_size = sizeof(more);
+    size_t index = 0;
+
+    while (true) {
+      rc = zmq_getsockopt(socket->socket_, ZMQ_EVENTS, &events, &events_size);
+      if (rc != 0) {
```

Comment:
Idioms serve their purpose. Usually you don't care about why a function succeeded, as long as it did, but if it failed, you want to know why, as there are many ways in which it failed. Since 0 is false and nonzero is true in C, it is natural API design to use 0 for success and an error code for the rest. There are not many other practical alternatives.

Personally, I hate dynamic typing and consider the notions of trueish and falsy to be anathema.

```js
if ( [] ) {
  // this runs
}

if ( [] == true ) {
  // this does not run
}

if ( [] == false ) {
  // this does run
}
```